### PR TITLE
add gitalk comment system

### PR DESCRIPTION
**Pull request by Jactor-Sue**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/iissnan/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes have been added (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [ ] Docs have been added / updated (for bug fixes / features).
    - I don't know where to update docs.

## PR Type
**What kind of change does this PR introduce?**  

- [ ] Bugfix.
- [x] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number(s): N/A

## What is the new behavior?
Description about this pull, in several words...

* Add Gitalk comment system according to blogs following:
  - [Hexo NexT主题中集成gitalk评论系统-asdfv1929](https://asdfv1929.github.io/2018/01/20/gitalk/)
  - [【已解决】Hexo NexT使用Gitalk未找到相关的Issues进行评论Error:Validation Failed-MoshiYeap](https://liujunzhou.cn/2018/8/10/gitalk-error/#more)
  - [在hexo中使用gitalk](https://github.com/gitalk/gitalk/wiki/%E5%9C%A8hexo%E4%B8%AD%E4%BD%BF%E7%94%A8gitalk)

* Demo as [Gitalk official site](https://gitalk.github.io/)

#### 1. Create new Github Application
Create a new Github application at [Register a new OAuth application](https://github.com/settings/applications/new), remember what you inputed, and then you will get `Client ID` and `Client Secret` used later.

#### 2. Enable Gitalk
Enable `gitalk` in NexT `_config.yml`, put `Client ID` and `Client Secret` down here:
```yml
gitalk:
  enable: true
  githubID:  # Github repo owner
  repo:  # Repository name to store issues.
  ClientID:  # Github Application Client ID
  ClientSecret:  # Github Application Client Secret
  adminUser:  # GitHub repo owner and collaborators, only these guys can initialize github issues
  distractionFreeMode: true # Facebook-like distraction free mode
...
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

**Comment by Jactor-Sue** [comment]
The file `md5.js` under `/source/js/src/` not third-party lib, source from https://github.com/blueimp/JavaScript-MD5 by `Sebastian Tschan` . This is used to hash url to fixed length string, and it works well in the tests for all themes.

**Comment by Jactor-Sue** [comment]
CDNs added for `gitalk` and `md5`.

**Comment by Jactor-Sue** [comment]
So the feature will not be added? Or other consideration?

**Comment by sli1989** [comment]
@Jactor-Sue options would be provided maybe. just waiting for review.

**Comment by maple3142** [comment]
Why you use `gitalk_cdn_prefix` instead of full url of gittalk script?
Aside from this, it will be better if it is place at `vendors.gittalk` and `vendors.md5`.
Because lots of cdn urls are already there, it will make `_config.yml` more consistent.

**Comment by Jactor-Sue** [comment]
Firstly, I used to use full url of scripts , but `gitalk_cdn_prefix` can be reused by `gitalk.css`, so I did this. I don't know which is better, I need an advice.

Solution1:
```html
<link rel="stylesheet" href=" {{ theme.gitalk.gitalk_cdn_prefix }}gitalk.css">
<script type="text/javascript" src="{{ theme.gitalk.gitalk_cdn_prefix }}gitalk.min.js"></script>
```
Solution2:
```html
<link rel="stylesheet" href=" {{ theme.gitalk.gitalk_cdn }}">
<script type="text/javascript" src="{{ theme.gitalk.gitalk_cdn }}"></script>
```
The second, I didn't catch what you mean about `vendors.gitalk` and `vendors.md5`, more specific please. 

I'm new here, thank you for your advice sincerely.

**Comment by maple3142** [comment]
[vendors](https://github.com/theme-next/hexo-theme-next/blob/1a87fc3cb4ecb0a8d69aada68c4bec9468bad5fe/_config.yml#L897) is a config entry where you can place cdn urls.
```yml
vendors:
  md5:
  gitalk:
  gitalk_css:
```

**Comment by Jactor-Sue** [comment]
@maple3142  Got it, I'm getting a commit, thanks.

File: _config.yml [hunk 1]
```diff
@@ -1009,6 +1021,19 @@ vendors:
   # Example: https://cdn.jsdelivr.net/npm/valine@1.1.8/dist/Valine.min.js
   valine:
 
+  # Internal version:
```
[the remark this answers is not in the document]
Comment (ivan-nginx) [reply _config.yml]: Yes, i think need to include internal version.

[the remark this answers is not in the document]
Comment (Jactor-Sue) [reply _config.yml]: Fixed, both `gitalk` and `md5` cdn source from jsDelivr now.

**Pull request merged**